java/code/src/com/redhat/rhn/domain/formula/FormulaFactory.java

```diff
@@ -1108,4 +1109,54 @@ public static String getClusterPillarDir() {
         return dataDir + PILLAR_DIR;
     }
 
+    /**
+     * Find endpoint information from given formula data
+     * @param formulaName name of the formula to examine
+     * @param formulaData formula data to extract information from
+     * @return list of endpoint information objects
+     */
+    public static List<EndpointInfo> getEndpointsFromFormulaData(String formulaName, FormulaData formulaData) {
+        return getExportersEndpoints(formulaData);
+    }
+
+    private static List<EndpointInfo> getExportersEndpoints(FormulaData formulaData) {
+        Map<String, Object> formulaValues = formulaData.getFormulaValues();
+        if (formulaValues.containsKey("exporters")) {
+            Boolean proxyEnabled = getValueByPath(formulaValues, "proxy_enabled")
+                    .filter(Boolean.class::isInstance)
+                    .map(Boolean.class::cast)
+                    .orElse(false);
+            Optional<Integer> proxyPort = proxyEnabled ? getValueByPath(formulaValues, "proxy_port")
+                    .filter(Number.class::isInstance)
+                    .map(Number.class::cast)
+                    .map(Number::intValue) : Optional.empty();
+            String proxyPath = proxyEnabled ? "/proxy" : null;
+            Boolean tlsEnabled = getValueByPath(formulaValues, "tls:enabled")
+                    .filter(Boolean.class::isInstance)
+                    .map(Boolean.class::cast)
+                    .orElse(false);
+
+            Map<String, Object> exportersMap = getValueByPath(formulaValues, "exporters")
+                    .filter(Map.class::isInstance)
+                    .map(Map.class::cast).get();
+
+            return exportersMap.entrySet().stream()
+                    .map(exporterEntry -> new ExporterConfig(exporterEntry.getKey(),
+                            Optional.ofNullable(exporterEntry.getValue())
+                                    .filter(Map.class::isInstance).map(Map.class::cast).get()))
```

Comment:
Not sure how big is the risk here, but could calling `get()` result in a NPE here (or on the previous assignment)?